client/my-sites/plugins/main.jsx

```diff
@@ -569,7 +569,7 @@ export default flow(
 				currentPluginsOnVisibleSites: getPlugins( state, visibleSiteIds, filter ),
 				pluginUpdateCount: pluginsWithUpdates && pluginsWithUpdates.length,
```

Comment:
Should we update all additional areas that fetch or return plugin data for the selected `siteIds` to actually do it for all sites? For example, `pluginsWithUpdates` will still return only the plugins with updates within the `siteIds` prop.

Reply:
In general, the `siteIds` in the UI and the sites returned from the backend should match, but I agree it would be better to update all additional areas as well to avoid any inconsistency.

What do you think about adding a new `getAllPlugins()` (or similar) selector that will be used when there's no selected site?

Reply:
Such a selector could solve the issue, but to me the root of the problem is that there are a bunch of components and subcomponents in the tree that potentially consume different sets of `siteId` or `siteIds` and that could lead to weird UX issues and errors.

Reply:
Initially, I thought we could replace all of them, but that would be a lot of work, and I'm not sure if we should include this in the scope of this PR.

Currently, we fetch the sites from the `me/sites` endpoint, then we have the [getSelectedOrAllSites](https://github.com/Automattic/wp-calypso/blob/7655b6d1f326371990851da10ebb97f133fbdc17/client/state/selectors/get-selected-or-all-sites-with-plugins.js#L16) selector that does some additional filtering.

The returned IDs from the above should match the ones returned _by the new endpoint_: active Jetpack sites where the current user is capable of managing plugins.

Can you think of any reason why the returned sites would differ in both situations?

Reply:
One example I can think of is this page: https://wordpress.com/plugins/manage

There you can still select one or more plugins to perform bulk actions on, and that would work only with the sites where that plugin is installed on. In that scenario, the list of sites can, and most likely will, differ from the list of all Jetpack sites.

Reply:
Actually, disregard this message. I realized that the state management for managing selected plugins for bulk editing is not handled in the main component, but rather by the `PluginsList`, a level down, so we should be good here. Seems like the above shouldn't be a concern 👍